src/compiler/scanner.ts

```diff
@@ -2343,8 +2399,12 @@ export function createScanner(languageVersion: ScriptTarget, skipTrivia: boolean
 
     function reScanSlashToken(): SyntaxKind {
         if (token === SyntaxKind.SlashToken || token === SyntaxKind.SlashEqualsToken) {
+            // Quickly get to the end of regex such that we know the flags
             let p = tokenStart + 1;
             let inEscape = false;
+            // Although nested character class is allowed in Unicode Sets mode,
+            // an unescaped slash is nevertheless invalid even in character class in Unicode mode.
+            // Thus we can simply ignore nested character class in the first pass.

```

Comment:
I guess what's happening is you're giving up the opportunity for a better error in cases like this:

```
var x = /[[\x00-\xff]&&\/]/v;
```

Could you at least add an example into the comment?

Reply:
It would be relatively easy to recover better if you just swapped `inCharacterClass` to be a stack counter.

Reply:
I once tried it but I can’t remember the rationale as to why I didn’t adopt it. I just tried it again and it really seems to give better error messages.

Reply:
I know why now. `/[[]/` is a totally valid regex. If we did that it would break all expressions like `/[[]/.test("[")`

Reply:
For the example you gave, `[/[[\x00-\xff]&&/]/v` is already a valid JavaScript expression if `v` is defined.